### PR TITLE
Implement `for` for signed integers

### PR DESCRIPTION
**Pull request by fee1-dead**
Best viewed commit-by-commit.

### CODE REVIEW

**Comment by Firestar99** [comment]
Thanks a lot for making this, cause I barely know anything about rustc dev and would have probably spend days trying to make something work :D

File: crates/rustc_codegen_spirv/src/builder/builder_methods.rs [hunk 1]
```diff
@@ -1318,28 +1318,80 @@ impl<'a, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'tcx> {
...
+            // when subtracting, overflow happens if
+            // - rhs is positive and result > lhs; or
+            // - rhs is negative and result < lhs
+            // this is equivalent to (rhs < 0) == (result < lhs)
+            let rhs_lt_zero = self.icmp(IntPredicate::IntSLT, rhs, self.constant_int(rhs.ty, 0));
+            let result_gt_lhs = self.icmp(
+                if is_add {
+                    IntPredicate::IntULT
+                } else {
+                    IntPredicate::IntSLT
```
Comment (fee1-dead) [inline crates/rustc_codegen_spirv/src/builder/builder_methods.rs]: this part is wrong,

Comment (fee1-dead) [reply crates/rustc_codegen_spirv/src/builder/builder_methods.rs]: should be fixed

**Comment by LegNeato** [comment]
This looks great, thanks!

I've invited you to our private maintainer zulip as you have more experience with rustc than the rest of us....feel free to decline if you aren't interested! We're also beefing up the number of maintainers so if you'd be interested ping me on zulip.

**Pull request merged**